syntax = "proto3";

option go_package = "go/petAdoption";

package petAdoption;

// The pet object definition
message Pet {
    int32 id = 1; // ID of the pet
    string name = 2; // Name of the pet
    string gender = 3; // Gender of the pet
    int32 age = 4; // Age of the pet
    string breed = 5; // Breed of the pet
    bytes picture = 6; // Picture of the pet
}

// Request message for inserting a pet
message InsertPetRequest {
    string name = 1;
    string gender = 2;
    int32 age = 3;
    string breed = 4;
    bytes picture = 5;
}

// Response message for insert operations
message InsertPetResponse {
    bool success = 1;
}

// Request message for searching pets
message SearchPetRequest {
    string searchTerm = 1;
}

// Response message for search results
message SearchPetResponse {
    repeated Pet pets = 1;
}

// The PetAdoption service definition
service PetAdoption {
    rpc InsertPet(InsertPetRequest) returns (InsertPetResponse);
    rpc SearchPet(SearchPetRequest) returns (SearchPetResponse);
}